cpp/src/strings/contains.cu

```diff
@@ -0,0 +1,244 @@
+                                     cudaStream_t stream = 0)
+{
+    return contains_util(strings, pattern, false, mr, stream);
+}
+
+std::unique_ptr<column> matches_re( strings_column_view const& strings,
+                                    std::string const& pattern,
+                                    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                    cudaStream_t stream = 0)
+{
+    return contains_util(strings, pattern, true, mr, stream);
+}
+
+} // namespace detail
+
+// external APIs
+
+std::unique_ptr<column> contains_re( strings_column_view const& strings,
+                                     std::string const& pattern,
+                                     rmm::mr::device_memory_resource* mr)
+{
+    return detail::contains_re(strings, pattern, mr);
+}
+
+std::unique_ptr<column> matches_re( strings_column_view const& strings,
+                                     std::string const& pattern,
+                                     rmm::mr::device_memory_resource* mr)
+{
+    return detail::matches_re(strings, pattern, mr);
+}
+
+namespace detail
+{
+
+namespace
+{
+
+/**
+ * @brief This counts the number of times the regex pattern matches in each string.
+ *
+ */
+template<size_t stack_size>
+struct count_fn
+{
+    reprog_device prog;
+    column_device_view d_strings;
+
+    __device__ int32_t operator()(unsigned int idx)
+    {
+        u_char data1[stack_size], data2[stack_size];
+        prog.set_stack_mem(data1,data2);
+        if( d_strings.is_null(idx) )
+            return 0;
+        string_view d_str = d_strings.element<string_view>(idx);
+        int32_t find_count = 0;
+        size_type nchars = d_str.length();
+        size_type begin = 0;
+        while( begin <= nchars )
+        {
+            auto end = nchars;
+            if( prog.find(idx,d_str,begin,end) <=0 )
+                break;
+            ++find_count;
+            begin = end > begin ? end : begin + 1;
+        }
+        return find_count;
+    }
+};
+
+}
+
+std::unique_ptr<column> count_re( strings_column_view const& strings,
+                                  std::string const& pattern,
+                                  rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                  cudaStream_t stream = 0)
+{
+    auto strings_count = strings.size();
+    auto strings_column = column_device_view::create(strings.parent(),stream);
+    auto d_column = *strings_column;
+
+    // compile regex into device object
+    auto prog = reprog_device::create(pattern,get_character_flags_table(),strings_count,stream);
+    auto d_prog = *prog;
+
+    // create the output column
+    auto results = make_numeric_column( data_type{INT32}, strings_count,
+        copy_bitmask( strings.parent(), stream, mr), strings.null_count(), stream, mr);
+    auto d_results = results->mutable_view().data<int32_t>();
```

Comment:
```suggestion
    auto d_results = results->mutable_view().data<size_type>();
```
Better to use the `size_type`

Reply:
> Better to use the `size_type`

But the `results` column here is literally of type `INT32`.
The `size_type` could change to `int64` or something else in the future.

Reply:
Ah, I missed it in the doc.